app/src/vanilla/java/cm/aptoide/pt/app/view/AppViewPresenter.java

```diff
@@ -0,0 +1,619 @@
+        })
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(__ -> {
+        }, err -> crashReport.log(err));
+  }
+
+  private void handleClickOnVideo() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> view.getScreenshotClickEvent())
+        .filter(event -> event.isVideo())
+        .doOnNext(videoClick -> {
+          appViewAnalytics.sendOpenVideoEvent();
+          appViewNavigator.navigateToUri(videoClick.getUri());
+        })
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(__ -> {
+        }, err -> crashReport.log(err));
+  }
+
+  private void handleClickOnDescriptionReadMore() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> view.clickedReadMore())
+        .doOnNext(readMoreClickEvent -> {
+          appViewAnalytics.sendReadMoreEvent();
+          appViewNavigator.navigateToDescriptionReadMore(readMoreClickEvent.getStoreName(),
+              readMoreClickEvent.getDescription(), readMoreClickEvent.getStoreTheme());
+        })
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(__ -> {
+        }, err -> crashReport.log(err));
+  }
+
+  private void handleClickOnDeveloperWebsite() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> view.clickDeveloperWebsite())
+        .flatMapSingle(__ -> appViewManager.getDetailedAppViewModel(appId, packageName))
+        .filter(app -> !TextUtils.isEmpty(app.getDetailedApp()
+            .getDeveloper()
+            .getWebsite()))
+        .doOnNext(app -> view.navigateToDeveloperWebsite(app.getDetailedApp()))
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(__ -> {
+        }, err -> crashReport.log(err));
+  }
+
+  private void handleClickOnDeveloperEmail() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> view.clickDeveloperEmail())
+        .flatMapSingle(__ -> appViewManager.getDetailedAppViewModel(appId, packageName))
+        .filter(app -> !TextUtils.isEmpty(app.getDetailedApp()
+            .getDeveloper()
+            .getEmail()))
+        .doOnNext(app -> view.navigateToDeveloperEmail(app.getDetailedApp()))
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(__ -> {
+        }, err -> crashReport.log(err));
+  }
+
+  private void handleClickOnDeveloperPrivacy() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> view.clickDeveloperPrivacy())
+        .flatMapSingle(__ -> appViewManager.getDetailedAppViewModel(appId, packageName))
+        .filter(app -> !TextUtils.isEmpty(app.getDetailedApp()
+            .getDeveloper()
+            .getPrivacy()))
```

Comment:
Does it make sense to filter out empty strings here? why is this section shown/clickable if it is empty?